cli/doctest_runner.rs

```diff
@@ -0,0 +1,358 @@
+          .filter_map(|tag_item| match tag_item.tag {
+            Tag::Example(ex_tag) => Some((
+              self.get_span_location(ex_tag.span),
+              ex_tag.text.value.to_string(),
+            )),
+            _ => None,
+          })
+      });
+    self.examples.borrow_mut().extend(examples);
+  }
+
+  fn check_var_decl(
+    &mut self,
+    var_decl: &VarDecl,
+    opt_export_decl: Option<&ExportDecl>,
+  ) {
+    var_decl.decls.iter().for_each(|decl| {
+      if let Some(expr) = &decl.init {
+        match &**expr {
+          Expr::Object(_) | Expr::Fn(_) | Expr::Class(_) | Expr::Arrow(_) => {
+            if let Some(export_decl) = opt_export_decl {
+              self.parse_span_comments(export_decl.span);
+            } else {
+              self.parse_span_comments(var_decl.span);
+            }
+          }
+          _ => {}
+        }
+      }
+    });
+  }
+}
+
+impl Visit for DocTestVisitor {
+  fn visit_class(&mut self, class: &Class, parent: &dyn Node) {
+    self.parse_span_comments(class.span);
+    swc_ecmascript::visit::visit_class(self, class, parent);
+  }
+
+  fn visit_function(&mut self, function: &Function, parent: &dyn Node) {
+    self.parse_span_comments(function.span);
+    swc_ecmascript::visit::visit_function(self, function, parent);
+  }
+
+  fn visit_var_decl(&mut self, var_decl: &VarDecl, parent: &dyn Node) {
+    self.check_var_decl(var_decl, None);
+    swc_ecmascript::visit::visit_var_decl(self, var_decl, parent);
+  }
+
+  fn visit_export_decl(&mut self, export_decl: &ExportDecl, parent: &dyn Node) {
+    match &export_decl.decl {
+      Decl::Var(var_decl) => self.check_var_decl(var_decl, Some(export_decl)),
+      Decl::Class(_) | Decl::Fn(_) => {
+        self.parse_span_comments(export_decl.span)
+      }
+      _ => {}
+    }
+    swc_ecmascript::visit::visit_export_decl(self, export_decl, parent);
+  }
+}
+
+pub async fn parse_jsdocs(
+  source_files: &[Url],
+  flags: Flags,
+) -> Result<Vec<(Location, String)>, AnyError> {
+  let program_state = ProgramState::new(flags.clone())?;
+  let loader = Box::new(program_state.file_fetcher.clone());
+
+  let doc_parser = DocParser::new(loader, false);
+  let mut results = vec![];
+  for url in source_files {
+    let source_code =
+      doc_parser.loader.load_source_code(&url.to_string()).await?;
+    let path = PathBuf::from(&url.to_string());
+    let media_type = MediaType::from(&path);
+    let specifier = ModuleSpecifier::resolve_url(&url.to_string())?;
+    let parsed_module = ast::parse(&specifier, &source_code, &media_type)?;
+    let mut doc_tester = DocTester::new(parsed_module);
+    results.extend(doc_tester.get_comments());
+  }
+  Ok(results)
+}
+
+pub fn prepare_doctests(
+  jsdocs: Vec<(Location, String)>,
+  fail_fast: bool,
+  quiet: bool,
+  filter: Option<String>,
+) -> Result<String, AnyError> {
+  let mut test_file =  "import * as assert from \"https://deno.land/std@0.70.0/testing/asserts.ts\";\n".to_string();
```

Comment:
Having `asserts` namespace available by default is probably useful, but @ry and I think we shouldn't do that for the first pass of this feature

Reply:
Okay will remove this